src/check-dependency-bumps.ts

```diff
@@ -0,0 +1,433 @@
+/**
+ * Dependency Bump Checker Script
+ *
+ * This script analyzes git diffs to find dependency version changes in package.json files.
+ * It focuses on dependencies and peerDependencies, excluding devDependencies.
+ *
+ */
+
+import type { WriteStream } from 'fs';
+import path from 'path';
+import { validateChangelogs, updateChangelogs } from './changelog-validator.js';
+import { getCurrentBranchName } from './repo.js';
+import { getStdoutFromCommand } from './misc-utils.js';
+import { getValidRepositoryUrl } from './project.js';
+import { readPackageManifest } from './package-manifest.js';
+import type { PackageInfo, PackageChanges } from './types.js';
+
+/**
+ * Retrieves the git diff between two references for package.json files.
+ *
+ * @param fromRef - The starting git reference (commit, branch, or tag).
+ * @param toRef - The ending git reference (commit, branch, or tag).
+ * @param projectRoot - The project root directory.
+ * @returns The raw git diff output.
+ */
+async function getGitDiff(
+  fromRef: string,
+  toRef: string,
+  projectRoot: string,
+): Promise<string> {
+  try {
+    return await getStdoutFromCommand(
+      'git',
+      [
+        'diff',
+        '-U9999', // Show maximum context to ensure section headers are visible
```

Comment:
Nit: 😅 sorry it took me way too long to understand what a "section header" was in this context. Here's a suggestion to maybe make it clearer

```suggestion
        '-U9999', // Show maximum context to ensure full dependency lists are visible
```